tests/sentry/explore/translation/test_alerts_translation.py

```diff
@@ -0,0 +1,675 @@
+from unittest.mock import patch
+
+import orjson
+import pytest
+from sentry_protos.snuba.v1.request_common_pb2 import TraceItemType
+from sentry_protos.snuba.v1.trace_item_attribute_pb2 import (
+    ExtrapolationMode as RPCExtrapolationMode,
+)
+from sentry_protos.snuba.v1.trace_item_attribute_pb2 import Function
+from sentry_protos.snuba.v1.trace_item_filter_pb2 import ComparisonFilter
+from urllib3.response import HTTPResponse
+
+from sentry.explore.translation.alerts_translation import (
+    rollback_detector_query_and_update_subscription_in_snuba,
+    snapshot_snuba_query,
+    translate_detector_and_update_subscription_in_snuba,
+)
+from sentry.incidents.grouptype import MetricIssue
+from sentry.incidents.models.alert_rule import (
+    AlertRuleDetectionType,
+    AlertRuleSeasonality,
+    AlertRuleSensitivity,
+)
+from sentry.incidents.utils.types import DATA_SOURCE_SNUBA_QUERY_SUBSCRIPTION
+from sentry.seer.anomaly_detection.store_data import SeerMethod
+from sentry.seer.anomaly_detection.types import StoreDataResponse
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.models import ExtrapolationMode, SnubaQueryEventType
+from sentry.testutils.cases import SnubaTestCase, TestCase
+from sentry.testutils.helpers.features import with_feature
+
+pytestmark = pytest.mark.sentry_metrics
+
+
+class AlertsTranslationTestCase(TestCase, SnubaTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.org = self.create_organization()
+        self.project = self.create_project(organization=self.org)
+
+    def test_snapshot_snuba_query_with_performance_metrics(self) -> None:
+        alert_rule = self.create_alert_rule(
+            organization=self.org,
+            projects=[self.project],
+            name="Test Alert",
+            query="transaction.duration:>100",
+            aggregate="count()",
+            time_window=10,
+            dataset=Dataset.PerformanceMetrics,
+        )
+        snuba_query = alert_rule.snuba_query
```

Comment:
For all these tests is the alert rule being created just for the purpose of making a snuba query to use later?As we're moving away from the alert rule world it'd be better to not write new code that uses them if possible - there is a test fixture `self.create_snuba_query()` that could be used to accomplish the same thing if I'm understanding the usage correctly - here's an example https://github.com/getsentry/sentry/blob/master/tests/sentry/incidents/test_logic.py#L2230-L2239